src/lexers/jsx-lexer.js

```diff
@@ -54,4 +58,59 @@ export default class JsxLexer extends HTMLLexer {
 
     return this.keys
   }
+
+  /**
+   * Recursively convert html tags and js injections to tags with the child index in it
+   *
+   * @param {string} string
+   *
+   * @returns string
+   */
+  eraseTags(string) {
+    const children = this.simplify(acorn.parse(string, {plugins: {jsx: true}}).body[0].expression.children, string);
```

Comment:
Please avoid line that are longer than 80 char. I would suggest doing this in two steps